flocker/node/script.py

```diff
@@ -174,6 +174,7 @@ def flocker_dataset_agent_main():
     # Later, construction of this object can be moved into
     # DatasetAgentServiceFactory.get_service where various options passed on
     # the command line could alter what is created and how it is initialized.
+    # XXX: Implement the factory proposed above (here or in a blocking / follow up issue)
```

Comment:
The implication being, I think, that this branch also needs to include a new configuration file (with user-facing documentation) and changes to packaging code so that the agent gets told about that configuration file when it gets started.

Which ... maybe is blocked by packaging work making it possible to start the block device agent *at all* (instead of the ZFS agent)?

Which is probably blocked by the last of Itamar's work to split the ZFS and container agents into two different command line scripts...